File: backend/Bioskopina.Services/Database/BioskopinaList.cs
using System;
using System.Collections.Generic;

namespace Bioskopina.Services.Database;

public partial class BioskopinaList
{
    public int Id { get; set; }

    public int ListId { get; set; }

    public int MovieId { get; set; }

    public virtual Bioskopina Movie { get; set; } = null!;

    public virtual List List { get; set; } = null!;
}

File: bioskopina_mobile/lib/models/bioskopina_list.dart
import 'package:json_annotation/json_annotation.dart';

import 'bioskopina.dart';

part 'bioskopina_list.g.dart';

@JsonSerializable()
class BioskopinaList {
  int? id;
  int? listId;
  int? movieId;
  Bioskopina? movie; // This is referencing the Bioskopina class

  // Constructor
  BioskopinaList(this.id, this.listId, this.movieId, this.movie);

  // Factory method to create an instance of BioskopinaList from JSON
  factory BioskopinaList.fromJson(Map<String, dynamic> json) =>
      _$BioskopinaListFromJson(json);

  // Method to convert an instance of BioskopinaList to JSON
  Map<String, dynamic> toJson() => _$BioskopinaListToJson(this);
}
